js/runner.go

```diff
@@ -318,9 +322,55 @@ func (r *Runner) SetOptions(opts lib.Options) error {
 		r.console = c
 	}
 
+	// FIXME: Resolver probably shouldn't be reset here...
+	// It's done because the js.Runner is created before the full
+	// configuration has been processed, at which point we don't have
+	// access to the DNSConfig, and need to wait for this SetOptions
+	// call that happens after all config has been assembled.
+	// We could make DNSConfig part of RuntimeOptions, but that seems
+	// conceptually wrong since the JS runtime doesn't care about it
+	// (it needs the actual resolver, not the config), and it would
+	// require an additional field on Bundle to pass the config through,
+	// which is arguably worse than this.
+	ttl, err := parseTTL(opts.DNS.TTL.String)
+	if err != nil {
+		return err
+	}
+	strategy := opts.DNS.Strategy.DNSStrategy
+	if !strategy.IsADNSStrategy() {
+		strategy = lib.DefaultDNSConfig().Strategy.DNSStrategy
+	}
+	r.Resolver = netext.NewResolver(r.ActualResolver, ttl, strategy)
+
 	return nil
 }
 
+func parseTTL(ttlS string) (time.Duration, error) {
+	ttl := time.Duration(0)
+	switch ttlS {
+	case "inf":
+		// cache "indefinitely"
+		ttl = time.Duration(math.MaxInt64)
```

Comment:
Hmm, this might be a problem :sweat_smile: In the check for cache freshness, you have `d.lastLookup.Add(r.ttl)`. But if `ttl` is already `math.MaxInt64`, then this will wrap around and become a negative number, so `time.Now().UTC()` will never be less than it, so we'll never use the cache. 

I think we should just use 1 year or something like that here, not `math.MaxInt64`

Reply:
Indeed, great catch, will change.

Reply:
Resolved by a825a9c4.